frontend/src/bundles/chat/pages/chat-modal.tsx

```diff
@@ -0,0 +1,72 @@
+import {
+    Modal,
+    ModalCloseButton,
+    ModalContent,
+    ModalOverlay,
+} from '~/bundles/common/components/components.js';
+import {
+    useAppDispatch,
+    useAppSelector,
+    useCallback,
+} from '~/bundles/common/hooks/hooks.js';
+
+import { Chat } from '../components/components.js';
+import { MessageSender } from '../enums/message-sender.js';
+import { actions as chatActions } from '../store/chat.js';
+import { type GenerateTextRequestDto, type Message } from '../types/types.js';
+
+type Properties = {
+    isChatOpen: boolean;
+    onModalChatClose: () => void;
+};
+
+const ChatModal: React.FC<Properties> = ({ isChatOpen, onModalChatClose }) => {
+    const dispatch = useAppDispatch();
+    const { messages } = useAppSelector(({ chat }) => ({
+        messages: chat.messages,
+    }));
+
+    const handleSendMessage = useCallback(
+        (payload: GenerateTextRequestDto) => {
+            const userMessage: Message = {
+                id: messages.length + 1,
+                sender: MessageSender.USER,
+                text: payload.message,
+                timeStamp: new Date(),
+            };
+
+            void dispatch(chatActions.addMessage(userMessage));
+            void dispatch(chatActions.sendMessage(payload));
```

Comment:
Right now you create and send ai message inside sendMessage reducer, but for user messages you have addMessage and sendMessage reducers. I think it would be better if you come up with a single approach